src/components/pages/Roles/forms/RoleFormTerms.tsx

```
@@ -243,74 +270,82 @@ function RoleTermExpiredTerms({
 }
 
 export default function RoleFormTerms() {
-  const [newTermIndex, setNewTermIndex] = useState<number>();
   const { t } = useTranslation('roles');
-  const { values } = useFormikContext<RoleFormValues>();
+  const { values, setFieldValue } = useFormikContext<RoleFormValues>();
+  const { getHat } = useRolesStore();
+
+  const roleHatTerms = useMemo(() => {
+    const roleFormHatId = values.roleEditing?.id;
+    const hat = getHat(roleFormHatId || '0x');
```

I feel like we usually trying to be more explicit about returning `undefined` and to not use these mock values like `0x` or `zeroAddress`
Though I don't have strong stand here - I'm personally ok-ish with approach of implicit `undefined` return

---

If possible it'd better to handle the "problem" that `roleFormHatId` is undefined (ie, throw, make sure it's defined before this execution point, etc), instead of relying on defaults, since we're currently tightening up our typing app-wide. Not sure what the context of this particular line is though

---

Current. I have since updated this. and is handled slightly differently. Sorry hadn't pushed up the latest and left over pseudo code.